src/ameba/formatter/json_formatter.cr

```diff
@@ -115,26 +128,30 @@ module Ameba::Formatter
       end_location : Crystal::Location?,
       message : String do
       def to_json(json)
-        json.object do
-          json.field :rule_name, rule_name
-          json.field :severity, severity
-          json.field :message, message
-          json.field :location,
-            {line: location.try &.line_number, column: location.try &.column_number}
-          json.field :end_location,
-            {line: end_location.try &.line_number, column: end_location.try &.column_number}
-        end
+        {
+          rule_name: rule_name,
+          severity:  severity,
+          message:   message,
+          location:  {
+            line:   location.try &.line_number,
+            column: location.try &.column_number,
+          },
+          end_location: {
+            line:   end_location.try &.line_number,
+            column: end_location.try &.column_number,
+          },
+        }.to_json(json)
```

Comment:
could you elaborate on the benefits of this change? Do we really need to create an intermediate hash object to make a json object?

Reply:
To be honest #1 is subjective. In my opinion, the previous version is readable. 

For #2 (consistency), the only way to guarantee it is to enforce it through a rule. Otherwise, (as it often happens) a new developer/maintainer comes and introduces discrepancies or tries to enforce other subjective rules re-writing the whole codebase. 

I'm open to do these changes, but just want to be sure the style is preserved in a long term (if we want to follow it).

Reply:
I agree with both comments.

> For #2 (consistency), the only way to guarantee it is to enforce it through a rule. Otherwise, (as it often happens) a new developer/maintainer comes and introduces discrepancies or tries to enforce other subjective rules re-writing the whole codebase.

While I do agree that would be the ideal solution, I suggest to leave it for a followup PR.